model/CSVDialog.cc

```diff
@@ -34,6 +53,111 @@ void CSVDialog::reportFromFile(const std::string& input, const std::string& outp
   reportFromCSVFile(is,of,spec);
 }
 
+// Return file name after downloading a CSV file from the web.
+std::string CSVDialog::loadWebFile(const std::string& url)
+{
+    try
+    {
+		// Parse input URL
+        boost::regex ex("(http|https)://([^/ :]+):?([^/ ]*)(/?[^ #?]*)\\x3f?([^ #]*)#?([^ ]*)");
```

Comment:
I'd probably write it as \\?? rather than \\x3f?
Also, raw strings are very useful for regexes, to avoid "leaning toothpick syndrome" 
R"(...)"

everything between the parentheses is interpreted literally - ", ' and \ have no special meaning.

Reply:
The regex does not correctly handle the case of a username and/or password being provided

Reply:
See my response two comments up.